libraries/engage/locations/components/FulfillmentSelector/FulfillmentSelectorReserve.jsx

```diff
@@ -0,0 +1,54 @@
+// @flow
+import React from 'react';
+import classNames from 'classnames';
+import { Grid } from '@shopgate/engage/components';
+import { i18n } from '../../../core';
+import { IN_STORE_PICKUP_LABEL, IN_STORE_PICKUP, ROPIS } from '../../constants';
+import { StockInfo } from '../StockInfo';
+import { ChangeLocationButton } from '../ChangeLocationButton';
+import { itemRow, itemColumn } from './FulfillmentSelectorItem.style';
+import { useFulfillmentSelectorState } from './FulfillmentSelector.hooks';
+import { container, unavailable } from './FulfillmentSelectorReserve.style';
+
+/**
+ * Renders the reservation item label.
+ * @returns {JSX}
+ */
+export function FulfillmentSelectorReserve() {
+  const {
+    location,
+    selectedLocation,
+    handleChange,
+    productFulfillmentMethods,
+  } = useFulfillmentSelectorState();
+  const usedLocation = selectedLocation || location;
+  const enabled = productFulfillmentMethods && productFulfillmentMethods.includes(ROPIS);
+
+  const handleChangeLocation = React.useCallback(() => {
+    handleChange(IN_STORE_PICKUP, true);
+  }, [handleChange]);
+
+  return (
+    <React.Fragment>
+      <div>
+        {i18n.text(IN_STORE_PICKUP_LABEL)}
+      </div>
+      {(enabled && usedLocation) && (
```

Comment:
Please check `isProductAvailable(location)` and show `locations.fulfillment.error.reserve` error message

Reply:
Done.